src/terms.jl

```diff
@@ -559,14 +559,19 @@ StatsBase.coefnames(t::InteractionTerm) =
 ################################################################################
 # old Terms features:
 
+hasintercept(f::FormulaTerm) = hasintercept(f.rhs)
 hasintercept(t::AbstractTerm) = InterceptTerm{true}() ∈ terms(t) || ConstantTerm(1) ∈ terms(t)
+omitsintercept(f::FormulaTerm) = omitsintercept(f.rhs)
 omitsintercept(t::AbstractTerm) =
     InterceptTerm{false}() ∈ terms(t) ||
     ConstantTerm(0) ∈ terms(t) ||
     ConstantTerm(-1) ∈ terms(t)
 
 hasresponse(t) = false
-hasresponse(t::FormulaTerm{LHS}) where {LHS} = LHS !== nothing
+hasresponse(t::FormulaTerm) =
+    t.lhs !== nothing && 
+    t.lhs != ConstantTerm(0) &&
+    t.lhs != InterceptTerm{false}
```

Comment:
These could all be `!==` I think.

Though the last is missing it's `()`

Reply:
Indeed.